### PR TITLE
Add background logic for deploy spring cloud app

### PR DESCRIPTION
**Pull request by Flanker32**
Add background logic for deploy spring cloud app

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/SpringCloudConfigurationType.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/SpringCloudConfigurationType.java:21]: remove the extra *

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeployConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+import static com.microsoft.intellij.runner.functions.AzureFunctionsConstants.NEED_SIGN_IN;
+
+public class SpringCloudDeployConfiguration extends AzureRunConfigurationBase<SpringCloudModel>
+        implements RunProfileWithCompileBeforeLaunchOption {
+
+    public static final String NEED_SPECIFY_MODULE = "Please specify module";
+    private final SpringCloudModel springCloudModel;
+
+    private Module module;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeployConfiguration.java]: use moduleName like function

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/SpringCloudConstants.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+package com.microsoft.intellij.runner.springcloud;
+
+public class SpringCloudConstants {
+
+    public static final String DISPLAY_NAME = "Azure Spring Cloud Services";
+    public static final int DEFAULT_PERSISTENT_DISK_SIZE = 50;
+    public static final String DEFAULT_DEPLOYMENT_NAME = "default";
+    public static final String DEFAULT_PERSISTENT_DISK_MOUNT_PATH = "/persistent";
+    public static final String AZURE_SPRING_CLOUD_ICON = "azure-springcloud-small";
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/SpringCloudConstants.java]: azure-springcloud-small.png?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeployConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Map;
+
+import static com.microsoft.intellij.runner.functions.AzureFunctionsConstants.NEED_SIGN_IN;
+
+public class SpringCloudDeployConfiguration extends AzureRunConfigurationBase<SpringCloudModel>
+        implements RunProfileWithCompileBeforeLaunchOption {
+
+    public static final String NEED_SPECIFY_MODULE = "Please specify module";
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/deploy/SpringCloudDeployConfiguration.java]: private

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                    .withUserAgent(CommonSettings.USER_AGENT)
+                    .withLogLevel(LogLevel.BODY_AND_HEADERS)
+                    .authenticate(AzureCliCredentials.create(), subscriptionId);
+            managerMap.put(subscriptionId, manager);
+        }
+        return managerMap.get(subscriptionId);
+    }
+
+    public static AppResourceInner execute(SpringCloudModel springCloudModel) throws IOException {
+        return null;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java]: where is 'execute' used?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        persistentDisk.withMountPath(SpringCloudConstants.DEFAULT_PERSISTENT_DISK_MOUNT_PATH)
+                .withSizeInGB(SpringCloudConstants.DEFAULT_PERSISTENT_DISK_SIZE);
+        return persistentDisk;
+    }
+
+    private static void uploadFileToStorage(File file, String sasUrl) throws IOException {
+        try {
+            final CloudFile cloudFile = new CloudFile(new URI(sasUrl));
+            cloudFile.uploadFromFile(file.getPath());
+        } catch (Exception e) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java]: no need to try-catch

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java (hunk 5 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/springcloud/library/SpringCloudUtils.java]: Has refactor this method which should be removed

**Review by andxu — APPROVED** [review]

**Pull request merged**